src/main/java/com/amazon/opendistroforelasticsearch/knn/plugin/script/KNNScoringUtil.java

```diff
@@ -34,18 +37,55 @@
     public static float l2Squared(float[] queryVector, float[] inputVector) {
         float squaredDistance = 0;
         for (int i = 0; i < inputVector.length; i++) {
-            float diff = queryVector[i]-inputVector[i];
+            float diff = queryVector[i] - inputVector[i];
             squaredDistance += diff * diff;
         }
         return squaredDistance;
     }
 
+    private static float[] toFloat(List<Number> inputVector) {
+        Objects.requireNonNull(inputVector);
+        float[] value = new float[inputVector.size()];
+        int index = 0;
+        for (final Number val : inputVector) {
+            value[index++] = val.floatValue();
+        }
+        return value;
+    }
+
+    /**
+     * Whitelisted l2Squared method for users to calculate L2 squared distance between query vector
+     * and document vectors
+     * Example
+     *  "script": {
+     *         "source": "1/(1 + l2Squared(params.query_vector, doc[params.field]))",
+     *         "params": {
+     *           "query_vector": [1, 2, 3.4],
+     *           "field": "my_dense_vector"
+     *         }
+     *       }
+     *
+     * @param queryVector query vector
+     * @param docValues   script doc values
+     * @return L2 score
+     */
+    public static float l2Squared(List<Number> queryVector, KNNVectorScriptDocValues docValues) {
+        float[] knnDocVector;
+        try {
+            knnDocVector = docValues.getValue();
+        } catch (Exception e) {
+            logger.debug("Failed to get vector from doc. Returning minimum score to put this result to end", e);
+            return Float.MIN_VALUE;
+        }
+        return l2Squared(toFloat(queryVector), knnDocVector);
+    }
+
     /**
      * This method can be used script to avoid repeated calculation of normalization
      * for query vector for each filtered documents
      *
-     * @param queryVector query vector
-     * @param inputVector input vector
+     * @param queryVector     query vector
```

Comment:
i believe that is done by the IntelliJ, to give better line of sight, instead of below,
int data;
float data1;
FooBar e;
the following will be easy for line of sight,
int         data2
float      data1
FooBar e
I observed this pattern in golang.

Reply:
Oh okay, makes sense. Yes, in the future I think we should use some kind of linter to keep style consistent.

Reply:
i agree.